crates/orderbook/src/api/post_quote.rs

```diff
@@ -281,29 +282,27 @@ impl OrderQuoter {
                     return Err(FeeError::PriceEstimate(PriceEstimationError::ZeroAmount));
                 }
 
-                let (fee, expiration) = self
-                    .fee_calculator
-                    .compute_subsidized_min_fee(
+                let price_estimation_query = price_estimation::Query {
```

Comment:
I see that there are common structures between `orderbook` and `shared` (`FeeData` and `price_estimation::Query`), is there a plan to extract these structures to common library? If not, should these have at least the same name, even if their current name reveals their usage intention?

Reply:
Nice observation! I personally think that while the fields are mostly overlapping on these two structs, they still have  different semantics that are kind of "specific" to their module. E.g. in the future the fee estimation query may depends on things like app data or user address (users with staked CoW token may get a fee discount) whereas the price estimation is likely never really depend on anything else but tokens, order type and amount.

I'd be all for aligning the naming and documenting the structs better though.